### PR TITLE
feat(APIv2): RHINENG-13519 expose the RuleTree on profiles/tailorings

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/policies/v2/tailoring_policy.rb [hunk 1]
```diff
@@ -11,6 +11,10 @@ def show?
       match_account?
     end
 
+    def rule_tree?
+      true
```
Comment (romanblanco) [inline app/policies/v2/tailoring_policy.rb:15]:
```suggestion
      match_account?
```

File: spec/integration/v2/tailorings_spec.rb [hunk 2]
```diff
@@ -195,6 +195,49 @@
...
+      tags 'Policies'
+      description 'Returns the Rule Tree of a Tailoring'
+      operationId 'TailoringRuleTree'
+      content_types
+
+      parameter name: :policy_id, in: :path, type: :string, required: true
+      parameter name: :tailoring_id, in: :path, type: :string, required: true
+
+      response '200', 'Returns the Rule Tree of a Tailoring' do
+        let(:tailoring_id) { item.id }
```
Comment (romanblanco) [inline spec/integration/v2/tailorings_spec.rb:222]: The `item` is not reused in the context of this endpoint.

```suggestion
        let(:tailoring_id) { FactoryBot.create(:v2_tailoring, :with_mixed_rules, policy: V2::Policy.find(policy_id), os_minor_version: 1).id }
```

Comment (skateman) [reply spec/integration/v2/tailorings_spec.rb:222]: What do you mean it is not reused? It IS reused...that's what I'm doing here

Comment (romanblanco) [reply spec/integration/v2/tailorings_spec.rb:222]: Under `/policies/{policy_id}/tailorings/{tailoring_id}/rule_tree`, the `item` is not being reused, it is only used once on the line 222.

This part is unnecessary to introduce:
```
    let(:item) do
      FactoryBot.create(:v2_tailoring, :with_mixed_rules, policy: V2::Policy.find(policy_id), os_minor_version: 1)
    end
```

Comment (skateman) [reply spec/integration/v2/tailorings_spec.rb:222]: No it is not, I am maintaining consistency across other integration (and non-integration) tests. The `item` variable is always used when retrieving a single entity or anything underneath.

**Pull request merged**